core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -103,8 +99,11 @@ private void convertMultiPhraseQuery(int currentPos, int[] termsIdx, MultiPhrase
             for (int i = 0; i < termsIdx.length; i++) {
                 queryBuilder.add(terms.get(i)[termsIdx[i]], pos[i]);
             }
-            PhraseQuery query = queryBuilder.build();
-            query.setBoost(orig.getBoost());
+            Query query = queryBuilder.build();
+            //nocommit not sure why we deal with boost here, given that we pass along the boost to flatten method too...
```

Comment:
I tend to agree, the fact that tests are green gives us a false yet comfortable sense of security :) If it wasn't highlighting stuff I would really look into writing the test that is missing, but...  I have no clue where to start